revocation/revocation.go

```diff
@@ -46,19 +69,36 @@ func New(httpClient *http.Client) (Revocation, error) {
 	}
 	return &revocation{
 		httpClient:       httpClient,
-		certChainPurpose: ocsp.PurposeCodeSigning,
+		certChainPurpose: x509.ExtKeyUsageCodeSigning,
 	}, nil
 }
 
-// NewTimestamp contructs a revocation object for timestamping certificate
-// chain
-func NewTimestamp(httpClient *http.Client) (Revocation, error) {
-	if httpClient == nil {
-		return nil, errors.New("invalid input: a non-nil httpClient must be specified")
+// Options specifies values that are needed to check revocation
+type Options struct {
+	// OCSPHTTPClient is the HTTP client for OCSP request. REQUIRED.
+	OCSPHTTPClient *http.Client
+
+	// CertChainPurpose is the purpose of the certificate chain. Supported
+	// values are x509.ExtKeyUsageCodeSigning and x509.ExtKeyUsageTimeStamping.
+	// REQUIRED.
+	CertChainPurpose x509.ExtKeyUsage
```

Comment:
If `CertChainPurpose` is not set, it is set to `0`, which is `x509.ExtKeyUsageAny`.

Reply:
Yes. This is in `revocation.go` file, under the `Options` struct, we require the user to provide this field, and the only supported values are `x509.ExtKeyUsageCodeSigning` and `x509.ExtKeyUsageTimeStamping` by design. Under `NewWithOptions`, if caller does not provide their option, it would fail.

Reply:
This is different from the `Options` under the `ocsp` package, where we cannot require the user to provide an option due to breaking change.